packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -389,6 +389,88 @@ it('should handle row clicks', async () => {
   expect(rows[2].classList).toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(3);
   expect(onRowClick).toHaveBeenCalledTimes(3);
+
+  await user.keyboard('[/ControlLeft][ShiftLeft>]'); // Release Control and Press Shift (without releasing it)
+  await user.click(getByText(mockedData()[0].name)); // Perform a click with `shiftKey: true`
+
+  expect(rows[0].classList).toContain('iui-selected');
+  expect(rows[1].classList).toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
+  expect(onSelect).toHaveBeenCalledTimes(3 + (1 + 3)); // # times before + (reset + # selected rows from start to end)
```

Comment:
After a shift click, `onSelect()` is currently called `1 time to reset all the rows to be unselected + number of selected/toggled rows from start to end (in this test, it is 3)`. Is it alright to call `onSelect()` each time a row is toggled? (i.e. referring to the second part of the above highlighted equation). Or should the second part of the above equation also be `1` (called once after toggling all rows)? I think that would be difficult to achieve so I feel the current way is alright, but just want to make sure.

Reply:
**[OUTDATED]**